src/main.rs

```diff
@@ -93,9 +132,43 @@ fn main() {
     let exercises = toml::from_str::<ExerciseList>(toml_str).unwrap().exercises;
     let verbose = matches.is_present("nocapture");
 
-    if matches.subcommand_matches("list").is_some() {
-        exercises.iter().for_each(|e| println!("{}", e.name));
+    // Handle the list command
+    if let Some(list_m) = matches.subcommand_matches("list") {
+        if ["paths", "names"].iter().all(|k| !list_m.is_present(k)) {
+            println!("{:<17}\t{:<46}\t{:<7}", "Name", "Path", "Status");
+        }
+        let filters = list_m.value_of("filter").unwrap_or_default().to_lowercase();
+        exercises.iter().for_each(|e| {
+            let fname = format!("{}", e.path.display());
+            let filter_cond = filters.split('|').filter(|f| f.len() > 0).any(|f| {
+                e.name.contains(&f) || fname.contains(&f)
+            });
+            let status = if e.looks_done() {
+                "Done"
+            } else {
+                "Pending"
+            };
+            let solve_cond = {
+                (status == "Done" && list_m.is_present("solved"))
```

Comment:
I would use the `looks_done` output in these conditions instead of comparing strings.

Reply:
Yes, that would be more idiomatic here. Done.